lib/wasi/src/runners/wasi_common.rs

```diff
@@ -26,14 +26,13 @@ impl CommonWasiOptions {
         container_fs: Arc<dyn FileSystem + Send + Sync>,
         wasi: &WasiAnnotation,
     ) -> Result<(), anyhow::Error> {
-        let fs = prepare_filesystem(&self.mapped_dirs, container_fs, |path| {
-            builder.add_preopen_dir(path).map_err(Error::from)
-        })?;
+        let fs = prepare_filesystem(&self.mapped_dirs, container_fs, builder)?;
 
         builder.add_preopen_dir("/")?;
-        if fs.read_dir(".".as_ref()).is_ok() {
-            // Sometimes "." won't be mounted so preopening will fail.
-            builder.add_preopen_dir(".")?;
+
+        if self.mapped_dirs.iter().all(|m| m.guest != ".") {
```

Comment:
What happens the user already specified a mount for `/` ? 
Seems like we should check for that and error out otherwise?

Reply:
It should be fine. This just makes sure `.` points to the root directory on startup and doesn't affect any existing mounts on `/`.